src/hubcast/clients/github/client.py

```diff
@@ -102,3 +102,27 @@ async def get_repo_config(self):
                 raise InvalidConfigYAMLError()
 
             return config
+
+    async def getitem(self, url: str):
+        """GET response from arbitrary GitHub API URL"""
+        gh_token = await self.auth.authenticate_installation(
+            self.repo_owner, self.repo_name
+        )
+
+        async with aiohttp.ClientSession() as session:
+            gh = gh_aiohttp.GitHubAPI(session, self.requester, oauth_token=gh_token)
+
+            return await gh.getitem(url)
+
+    async def get_prs(self, branch=None):
+        """Get all open PRs in the repository, can be filtered by internal branches."""
+
+        # https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests
+        # default is open pull requests
+
+        url = f"/repos/{self.repo_owner}/{self.repo_name}/pulls"
+        if branch:
+            # head: filter pulls by head user or head organization and branch name
+            url = f"{url}?head={self.repo_owner}:{branch}"
+
+        return await self.getitem(url)
```

Comment:
Do we need to do any cleaning of the response here? It'd be good to just return a list of pr numbers instead of the full json response.